tests/unit/test_plots.py

```diff
@@ -191,77 +185,86 @@ def test_posterior_plots(self, posterior_summary):
+        builder = pybop.Pybamm()
+        builder.set_simulation(model)
+        builder.set_dataset(dataset)
+        builder.add_parameter(
             pybop.Parameter(
                 "Negative electrode active material volume fraction",
                 prior=pybop.Gaussian(0.68, 0.05),
                 bounds=[0.5, 0.8],
-            ),
+            )
+        )
+        builder.add_cost(
+            pybop.costs.pybamm.SumSquaredError("Voltage [V]", "Voltage [V]", 1)
         )
-        fitting_problem = pybop.FittingProblem(model, parameters, dataset)
-        cost = pybop.SumSquaredError(fitting_problem)
+        fitting_problem = builder.build()
         with pytest.raises(
             ValueError, match="This cost function takes fewer than 2 parameters."
         ):
-            pybop.plot.contour(cost)
+            pybop.plot.contour(fitting_problem, steps=5)
 
         # Test with more than two paramters
-        parameters = pybop.Parameters(
-            pybop.Parameter(
-                "Negative electrode active material volume fraction",
-                prior=pybop.Gaussian(0.68, 0.05),
-                bounds=[0.5, 0.8],
-            ),
+        builder.add_parameter(
             pybop.Parameter(
                 "Positive electrode active material volume fraction",
                 prior=pybop.Gaussian(0.58, 0.05),
                 bounds=[0.4, 0.7],
-            ),
+            )
+        )
+        builder.add_parameter(
             pybop.Parameter(
                 "Positive particle radius [m]",
                 prior=pybop.Gaussian(4.8e-06, 0.05e-06),
                 bounds=[4e-06, 6e-06],
-            ),
+            )
         )
-        fitting_problem = pybop.FittingProblem(model, parameters, dataset)
-        cost = pybop.SumSquaredError(fitting_problem)
-        pybop.plot.contour(cost)
+        fitting_problem = builder.build()
+        pybop.plot.contour(fitting_problem, steps=5)
 
     def test_contour_prior_bounds(self, model, dataset):
-        # Test with prior bounds
-        parameters = pybop.Parameters(
+        builder = pybop.Pybamm()
+        builder.set_simulation(model)
+        builder.set_dataset(dataset)
+        builder.add_parameter(
             pybop.Parameter(
                 "Negative electrode active material volume fraction",
                 prior=pybop.Gaussian(0.68, 0.01),
-            ),
+            )
+        )
+        builder.add_parameter(
             pybop.Parameter(
                 "Positive electrode active material volume fraction",
                 prior=pybop.Gaussian(0.58, 0.01),
-            ),
+            )
+        )
+        builder.add_cost(
+            pybop.costs.pybamm.NegativeGaussianLogLikelihood(
+                "Voltage [V]", "Voltage [V]", sigma=1e-3
+            )
         )
-        fitting_problem = pybop.FittingProblem(model, parameters, dataset)
-        cost = pybop.SumSquaredError(fitting_problem)
+
+        cost = builder.build()
```

Comment:
Same as previous, updating to "problem" would make this clearer.
```suggestion
        problem = builder.build()
```